### PR TITLE
DAOS-8025 control,bio: disable dpdk telemetry

### PR DESCRIPTION
**Pull request by tanabarr**
DPDK exposes telemetry through a temporary socket file.
To avoid telemetry socket file conflicts between runs with different
users, disable DPDK telemetry by passing option override when
initializing SPDK environment. DAOS doesn't use DPDK telemetry.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/bio/bio_xstream.c [hunk 1]
```diff
@@ -83,18 +83,27 @@ bio_spdk_env_init(void)
...
+
 	/*
 	 * TODO: Set opts.mem_size to nvme_glb.bd_mem_size
 	 * Currently we can't guarantee clean shutdown (no hugepages leaked).
 	 * Setting mem_size could cause EAL: Not enough memory available error,
 	 * and DPDK will fail to initialize.
 	 */
 
-	if (nvme_glb.bd_shm_id != DAOS_NVME_SHMID_NONE)
+	if (nvme_glb.bd_shm_id == DAOS_NVME_SHMID_NONE)
```
Comment (NiuYawei) [inline src/bio/bio_xstream.c:94]: I don't follow this change. The opts.shm_id is already initialized as -1 above.

File: src/bio/bio_xstream.c [hunk 2]
```diff
@@ -83,18 +83,27 @@ bio_spdk_env_init(void)
...
+	/*
+	 * TODO: Find a way to set multiple overrides, currently only single
+	 * option can be overridden with opts.env_context.
+	 */
+
+	/*
+	 * Quiet DPDK logging by setting level to ERROR
+	 *  opts.env_context = "--log-level=lib.eal:4";
+	 */
+	opts.env_context = "--no-telemetry";
```
Comment (NiuYawei) [inline src/bio/bio_xstream.c:106]: opts.env_context = "--log-level=lib.eal:4 --no-telemetry" doesn't work?

File: src/bio/bio_xstream.c (hunk 1 above)
Comment (tanabarr) [reply src/bio/bio_xstream.c]: fixed, thanks

File: src/bio/bio_xstream.c (hunk 2 above)
Comment (tanabarr) [reply src/bio/bio_xstream.c:106]: no, only a single option can be overridden using env_context, details added to the ticket

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**